cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/TemplateProcessingTest.java

```diff
@@ -80,13 +68,19 @@ public void cleanup() {
   @Test
   public void extractTemplateSetsTest() throws IOException {
 
-    FileUtils.copyDirectory(new File(testFileRootPath + "templates"),
-        this.cobiGenHome.resolve("template-sets/downloaded").toFile());
-    CobiGenFactory.extractTemplates();
-    Path adaptedFolder = this.cobiGenHome.resolve(ConfigurationConstants.CONFIG_PROPERTY_TEMPLATE_SETS_PATH)
-        .resolve(ConfigurationConstants.ADAPTED_FOLDER);
-    Path extractedJar1 = adaptedFolder.resolve("template-test1-0.0.1");
-    Path extractedJar2 = adaptedFolder.resolve("template-test2-0.0.1");
+    FileUtils.copyDirectory(new File(testFileRootPathTemplateSets), this.cobiGenHomeTemplateSets.toFile());
+
+    Path templateSetsFolder = this.cobiGenHomeTemplateSets
+        .resolve(ConfigurationConstants.CONFIG_PROPERTY_TEMPLATE_SETS_PATH);
+    Path downloadedFolder = templateSetsFolder.resolve(ConfigurationConstants.DOWNLOADED_FOLDER);
+    Path adaptedFolder = templateSetsFolder.resolve(ConfigurationConstants.ADAPTED_FOLDER);
+
+    TemplateAdapter templateAdapter = new TemplateAdapterImpl(downloadedFolder.getParent());
+    List<Path> templates = templateAdapter.getTemplateSetJars();
+    templateAdapter.adaptTemplateSets(templates, adaptedFolder, false);
+
+    Path extractedJar1 = adaptedFolder.resolve("template-test1-0.0.1/src/main/templates");
+    Path extractedJar2 = adaptedFolder.resolve("template-test2-0.0.1/src/main/templates");
```

Comment:
Please use: ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER instead of the fixed string.